superset/views/datasource/schemas.py

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Any
+
+from marshmallow import fields, post_load, Schema
+from typing_extensions import TypedDict
+
+
+class ExternalMetadataParams(TypedDict):
```

Comment:
@villebro @zhaoyongjie I wonder if we should be using [marshmallow_dataclass](https://github.com/lovasoa/marshmallow_dataclass) instead as it adhere's to the DRY principle and simplifies the code. I've used it for an internal project and I'm definitely a fan.

Reply:
Thanks for the mention. It's a more elegant way, let me try to use `marshmallow_dataclass` to refactor this class.